### DISCUSSION TITLE
Data Importer: Deleted transactions causing duplicate errors

### DISCUSSION THREAD

**Discussion by jtcooper10** [firefly-iii/firefly-iii#8490] [Firefly III]
Hi! I ran into an issue when trying to reattempt a data importer job, but didn't see any open issues for it or discussion. Unsure if this is me misunderstanding something, or if it's a bug.

Earlier today, I submitted a .csv with roughly 100 records or so incorrectly. I could have gone through and fixed them one by one but that would've taken forever, so I ended up just going to Classification > Tags, choosing the new import tag, and deleting all those transactions and then the tag itself, so that I can just start over from scratch.

But now, anytime I try to import that same file, no transactions ever get imported. The firefly-iii logs indicate that the hashes matched an existing transaction, and by extension the data-importer logs indicate that every transaction in the .csv is a duplicate. But the "duplicate" transactions were the ones that were deleted from the tag page.

I confirmed that they are no longer there, both in Firefly and in the API (e.g. for the above issue, querying `/v1/api/transactions/1061` returned a "Resource not found" error). Are deleted transactions still considered in duplicate detection (i.e. they're not actually "deleted" but just "flagged" for deletion), and if so is there a better way for me to "undo" transactions to avoid this? Or is there maybe a cache somewhere that I can "flush"? Etc.

I was able to work around this by temporarily turning off duplicate detection, but if I did something incorrectly or misunderstood something, I'd like to know how I can avoid this in the future.

Thanks!

**Comment by JC5** [comment]
https://docs.firefly-iii.org/how-to/data-importer/import/duplicates/

**Reply by jtcooper10** [reply]
Ah that's embarrassing, turns out I'm just blind. Thanks!
